spock-core/src/main/java/org/spockframework/runtime/IDataIterator.java

```diff
@@ -0,0 +1,24 @@
+package org.spockframework.runtime;
+
+import org.spockframework.util.Beta;
+
+import java.util.*;
+
+/**
+ * A special iterator, that gives to the data produced by Spock's data providers.
+ *
+ * @author Leonard Brünings
+ * @since 2.2
+ */
+@Beta
+public interface IDataIterator extends Iterator<Object[]>, AutoCloseable {
```

Comment:
🤔 I think it would be good to document who's responsible for closing this iterator, i.e. that the `IDataDriver` must _not_.

Reply:
done [1c18e7c](https://github.com/spockframework/spock/pull/1479/commits/1c18e7c1e230a08fb207292436d8e892bb04851b)